### PR TITLE
TestData: Adds important new features to the random walk scenario

### PR DESCRIPTION
**Pull request by torkelo**
Random walk now supports:
* Series count (generate multiple series from one query) 
* start value, spread and noise, min & max
* automatic series names based on different types (server names,  house locations)

### CODE REVIEW

File: public/app/plugins/datasource/testdata/partials/query.editor.html [hunk 1]
```diff
@@ -48,6 +48,68 @@
...
+				step="1"
+				ng-change="ctrl.refresh()" />
+		</div>
+    <div class="gf-form">
+			<label class="gf-form-label query-keyword width-7">Start value</label>
+			<input type="number"
+				class="gf-form-input width-6"
+				placeholder="auto"
+				ng-model="ctrl.target.startValue"
+				min="1"
```
Comment (papagian) [inline public/app/plugins/datasource/testdata/partials/query.editor.html]: Why the start value cannot be lower that `1`?

File: public/app/plugins/datasource/testdata/partials/query.editor.html [hunk 2]
```diff
@@ -48,6 +48,68 @@
...
+				step="0.1"
+				ng-change="ctrl.refresh()" />
+		</div>
+    <div class="gf-form">
+			<label class="gf-form-label query-keyword width-7">Min</label>
+			<input type="number"
+				class="gf-form-input width-6"
+				placeholder="none"
+				ng-model="ctrl.target.min"
+				min="0"
```
Comment (papagian) [inline public/app/plugins/datasource/testdata/partials/query.editor.html]: why `Min` cannot be negative?

File: public/app/plugins/datasource/testdata/partials/query.editor.html [hunk 3]
```diff
@@ -48,6 +48,68 @@
...
+				step="0.1"
+				ng-change="ctrl.refresh()" />
+		</div>
+    <div class="gf-form">
+			<label class="gf-form-label query-keyword width-7">Max</label>
+			<input type="number"
+				class="gf-form-input width-6"
+				placeholder="none"
+				ng-model="ctrl.target.max"
+				min="0"
```
Comment (papagian) [inline public/app/plugins/datasource/testdata/partials/query.editor.html]: why `Max` cannot be negative?

File: pkg/tsdb/testdatasource/scenarios.go [hunk 4]
```diff
@@ -520,59 +534,62 @@ func predictableSeries(timeRange *tsdb.TimeRange, timeStep, length int64, getVal
...
 	timeWalkerMs := tsdbQuery.TimeRange.GetFromAsMsEpoch()
 	to := tsdbQuery.TimeRange.GetToAsMsEpoch()
+	series := newSeriesForQuery(query, index)
 
-	series := newSeriesForQuery(query)
+	startValue := query.Model.Get("startValue").MustFloat64(rand.Float64() * 100)
+	spread := query.Model.Get("spread").MustFloat64(1)
+	noise := query.Model.Get("noise").MustFloat64(0)
+	min, hasMin := query.Model.Get("min").Float64()
+	max, hasMax := query.Model.Get("max").Float64()
```
Comment (papagian) [inline pkg/tsdb/testdatasource/scenarios.go]: I would change this to:
```
min, err := query.Model.Get("min").Float64()
hasMin := err == nil
max, err := query.Model.Get("max").Float64()
hasMax := err == nil
```
so the conditionals below would become:
`if hasMin && nextValue < min` and
`if hasMax && nextValue > max`

**Review by marefr — APPROVED** [review]
LGTM

**Review by papagian — APPROVED** [review]
It works nice; I just wonder why the `start`, `min` and `max` cannot get negative values.

**Pull request merged**